src/ServiceControl/Recoverability/API/FailureGroupsApi.cs

```diff
@@ -27,8 +27,13 @@ public FailureGroupsApi()
             Post["/recoverability/groups/reclassify"] =
                 _ => ReclassifyErrors();
 
-            Get["/recoverability/groups/{classifier?Exception Type and Stack Trace}/{classifierFilter?}"] =
-                parameters => GetAllGroups(parameters.Classifier, parameters.classifierFilter == "undefined" ? null : parameters.classifierFilter);
+            Get["/recoverability/groups/{classifier?Exception Type and Stack Trace}"] =
+               parameters =>
+               {
+                   var classifierFilter = Request.Query["classifierFilter"] == null || Request.Query["classifierFilter"] == "undefined" ? null : Request.Query["classifierFilter"];
```

Comment:
Do we need the first `Request.Query["classifierFilter"] == null` check? 

If I understand correctly this
```
var classifierFilter = Request.Query["classifierFilter"] == null || Request.Query["classifierFilter"] == "undefined" ? null : Request.Query["classifierFilter"];
```
means `classifierFilter` will be either some value of `null` if `classifierFilter` query string param is not present or its value is `"undefined"`.

This should be equivalent:

```
var classifierFilter = Request.Query["classifierFilter"] == "undefined" ? null : Request.Query["classifierFilter"];
```

or even better

```
var classifierFilter = Request.Query["classifierFilter"] != "undefined" ? Request.Query["classifierFilter"] : null;
```